src/sgl/utils/texture_loader.cpp

```diff
@@ -127,8 +128,12 @@ inline std::pair<Format, bool> determine_texture_format(const Bitmap* bitmap, co
     // Check if bitmap is RGB and we can convert to RGBA.
     bool convert_to_rgba = false;
     if (options.extend_alpha && pixel_format == PixelFormat::rgb) {
-        bool rgb_format_supported
-            = FORMAT_TABLE.find(make_key(PixelFormat::rgb, component_type, format_flags)) != FORMAT_TABLE.end();
+        // Find if the RGB format exists, if it does check if the device supports it
+        auto rgb_it = FORMAT_TABLE.find(make_key(PixelFormat::rgb, component_type, format_flags));
+        bool rgb_format_supported = rgb_it != FORMAT_TABLE.end();
+        if (rgb_format_supported && device->get_format_support(rgb_it->second) != FormatSupport::texture)
```

Comment:
you probably want `is_set(format_support, FormatSupport::texture)` here (mulitple flags could be set)